x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import createContainer from 'constate';
+import { useCallback, useReducer } from 'react';
+import { buildEsQuery, Filter, Query, TimeRange } from '@kbn/es-query';
+import DateMath from '@kbn/datemath';
+import type { SavedQuery } from '@kbn/data-plugin/public';
+import type { InfraClientStartDeps } from '../../../../types';
+import { useMetricsDataViewContext } from './use_data_view';
+import { useKibanaTimefilterTime } from '../../../../hooks/use_kibana_timefilter_time';
+
+const DEFAULT_FROM_MINUTES_VALUE = 15;
+
+export const useUnifiedSearch = () => {
+  const [, forceUpdate] = useReducer((x: number) => x + 1, 0);
```

Comment:
This is what react recommends https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate. But this is temporary. I think the URL state ticket will force the component to update instead of this.

Reply:
Thanks for the explanation! We can keep it, you are right that when we update the url state and set the query/filters/time range it should be fine without it but we will probably still need to force the update when saving the query for example.

Reply:
Perhaps we don't even need that `forceUpdate` in the `saveQuery` function. It's not supposed to trigger a new search. Let me double check that.